ebean-core/src/main/java/io/ebeaninternal/server/deploy/BeanPropertyJsonMapper.java

```diff
@@ -43,8 +44,8 @@ public Object readSet(DataReader reader, EntityBean bean) throws SQLException {
         setValue(bean, value);
         String json = reader.popJson();
         if (json != null) {
-          final String hash = Md5.hash(json);
-          bean._ebean_getIntercept().mutableHash(propertyIndex, hash);
+          final String hash = scalarType.format(value);
+          bean._ebean_getIntercept().mutableHash(propertyIndex, scalarType.jsonMutable(hash));
```

Comment:
This seems to be suboptimal. We can use "json" directly here
```suggestion
          bean._ebean_getIntercept().mutableHash(propertyIndex, scalarType.jsonMutable(json));
```